Classes/Executor/OptimizationExecutorRemoteImageoptim.php

```diff
@@ -30,21 +30,105 @@
 
 class OptimizationExecutorRemoteImageoptim extends OptimizationExecutorRemote
 {
+
     /**
-     * Optimize image using remote Imageoptim
-     * Return the temporary file path
+     * Optimize image
      *
-     * @param string $inputImageAbsolutePath Absolute path/file with image to be optimized. It will be replaced with optimized version.
-     * @param Configurator $configurator Executor configurator
-     * @return ExecutorResult Executor Result
+     * @param $inputImageAbsolutePath string Absolute path/file with image to be optimized
+     * @param Configurator $configurator
+     * @return ExecutorResult Optimization result
      */
     public function optimize(string $inputImageAbsolutePath, Configurator $configurator) : ExecutorResult
     {
         $executorResult = GeneralUtility::makeInstance(ExecutorResult::class);
         $executorResult->setExecutedSuccessfully(false);
 
-        // Implement optimize image with Imageoptim and fill all $execuroeResult fields
+        if (!empty($configurator->getOption('api.key'))) {
+            $executorResult->setSizeBefore(filesize($inputImageAbsolutePath));
+            $this->initialize([
+                'auth' => [
+                    'api_key' => $configurator->getOption('api.key'),
+                ],
+                'url'  => [
+                    'upload' => 'https://im2.io',
```

Comment:
Please make that option configurable in default TSConfig for given provider. 
We should have as low hardcoded values as possible.